source/features/set-default-repositories-type-to-sources.tsx

```diff
@@ -1,25 +1,34 @@
 import select from 'select-dom';
 
 import features from '.';
+import onFragmentLoad from '../github-events/on-fragment-load';
 
-function init(): void {
+function addSourceTypeToLink(link: HTMLAnchorElement): void {
+	const search = new URLSearchParams(link.search);
+	search.set('type', 'source');
+	link.search = String(search);
+}
+
+function headerDropdownListener(): void {
+	addSourceTypeToLink(select<HTMLAnchorElement>('.header-nav-current-user ~ [href$="tab=repositories"]')!);
+}
+
+async function init(): Promise<void> {
 	const links = select.all<HTMLAnchorElement>([
 		// Pre "Repository refresh" layout
-		'.user-profile-nav [href$="tab=repositories"]', // "Your repositories" in header dropdown
 		'#user-links [href$="tab=repositories"]', // "Repositories" tab on user profile
 		'.orgnav > .pagehead-tabs-item:first-child', // "Repositories" tab on organization profile
 		// "Repository refresh" layout
-		'.header-nav-current-user ~ [href$="tab=repositories"]', // "Your repositories" in header dropdown
 		'[aria-label="User profile"] [href$="tab=repositories"]', // "Repositories" tab on user profile
 		'[aria-label="Organization"] .UnderlineNav-item:first-child', // "Repositories" tab on organization profile
 		'[data-hovercard-type="organization"]' // Organization name on repo header + organization list on user profile
 	]);
 
 	for (const link of links) {
-		const search = new URLSearchParams(link.search);
-		search.set('type', 'source');
-		link.search = String(search);
+		addSourceTypeToLink(link);
 	}
+
+	onFragmentLoad('.Header-item > details include-fragment', headerDropdownListener); // "Your repositories" in header dropdown
```

Comment:
I made the wrong suggestion in this case since this fragment only ever loads **once**, unlike the "PR file load" listener.

We only need a promise in this case:

```ts
// If the dropdown exists, await it; If not, it probably already loaded
async function onProfileDropdownLoad(): Promise<void> {
	const dropdown = select('.Header details-menu[src^="/users/"] include-fragment');
	if (dropdown) {
		await oneEvent(dropdown, 'load');
	}
}
```
```suggestion
	// "Your repositories" in header dropdown
	await onProfileDropdownLoad();
	addSourceTypeToLink(select<HTMLAnchorElement>('.header-nav-current-user ~ [href$="tab=repositories"]')!);
```

The good part in this case is that if the code runs _after_ the fragment has been loaded, it still works.

Possible improvements **in a new PR**

- add `on-fragment-load.ts` (you can remove it from this PR)
- import it inside `on-pr-file-load` and use it inside
- look for other features that expect/await `include-fragment` and see whether they can use either `on-fragment-load` or `onProfileDropdownLoad`